test/asserts.go

```diff
@@ -49,6 +49,19 @@ func AssertNotNil(t *testing.T, obj interface{}, message string) {
 	}
 }
 
+// AssertBoxedNil checks that an inner object is nil. This is intentional for
+// testing purposes only.
+func AssertBoxedNil(t *testing.T, obj interface{}, message string) {
+	t.Helper()
+	switch reflect.TypeOf(obj).Kind() {
+	// .IsNil() only works on chan, func, interface, map, pointer, and slice.
+	case reflect.Chan, reflect.Func, reflect.Interface, reflect.Map, reflect.Pointer, reflect.Slice:
+		if !reflect.ValueOf(obj).IsNil() {
+			t.Fatal(message)
+		}
+	}
```

Comment:
You probably want a `default:` case here, so that the function can _also_ fail if the thing it is passed isn't an interface at all.